GW2EIParser/MainForm.cs

```diff
@@ -877,4 +878,24 @@ private void EnableSettingsWatcher(object sender, EventArgs e)
         AddTraceMessage("Settings: Closing settings");
         BtnSettings.Enabled = true;
     }
+
+    private void BtnCheckUpdates_Click(object sender, EventArgs e)
+    {
+        AddTraceMessage("Updater: Checking for updates");
+        Updater updater = new();
+
+        var mainForm = this;
+        var uiScheduler = TaskScheduler.FromCurrentSynchronizationContext();
+        Task.Factory.StartNew(async () =>
+        {
+            if (await updater.NewReleaseCheckerAsync())
+            {
+                mainForm.Invoke(() => // Must run on UI thread
+                {
+                    var updaterForm = new UpdaterForm(updater);
+                    updaterForm.Show();
```

Comment:
ShowDialog(mainForm) to disable interactions on the main form while the update window is open.